### PR TITLE
Merge `CODEOWNERS` Part 1

### LINKED ISSUE #1164: Cleanup: CODEOWNERS file
Codeowner files need to have their paths updated. [GH does not mention](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location) to support multiple of those, so we need to merge them.

### PR DESCRIPTION
**Pull request by chevdor**
This PR merged the previous `CODEOWNERS` files.

fix #1164 

See also #1415

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@franciscoaguirre I added polkadot-sdk to the xcm team.

**Comment by franciscoaguirre** [comment]
@bkchr I added the rule to the file, but it's still giving an error.
I think it's because we have only "read" access, we need "write" according to GitHub.

image

**Comment by bkchr** [comment]
> @bkchr I added the rule to the file, but it's still giving an error.
> I think it's because we have only "read" access, we need "write" according to GitHub.
> 
> image
> 

Fixed.

**Review by kianenigma — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# - Multiple owners are supported.
+# - Either handle (e.g, @github_user or @github/team) or email can be used. Keep in mind,
+#   that handles might work better because they are more recognizable on GitHub,
+#   eyou can use them for mentioning unlike an email.
+# - The latest matching rule, if multiple, takes precedence.
+
+# CI
+/.github/ @paritytech/ci @paritytech/release-engineering
+/.gitlab-ci.yml @paritytech/ci
+/scripts/ci/ @paritytech/ci @paritytech/release-engineering
```
Comment (alvicsam) [inline .github/CODEOWNERS:22]:
```suggestion
/.gitlab/ @paritytech/ci
```
Currently there is no `scripts` folder and I'm not sure that we should return it back.

**Review by alvicsam — CHANGES_REQUESTED** [review]

**Review by alvicsam — APPROVED** [review]

**Pull request merged**